domain/src/main/kotlin/no/nav/su/se/bakover/domain/vilkår/FlyktningVilkår.kt

```diff
@@ -0,0 +1,233 @@
+                    }
+                }
+                Behandlingsinformasjon.Flyktning.Status.Uavklart -> IkkeVurdert
+            }
+        }
+    }
+
+    object IkkeVurdert : FlyktningVilkår() {
+        override val resultat: Resultat = Resultat.Uavklart
+        override val erAvslag = false
+        override val erInnvilget = false
+        override val grunnlag = emptyList<FlyktningGrunnlag>()
+        override fun lagTidslinje(periode: Periode): FlyktningVilkår {
+            return this
+        }
+
+        override fun oppdaterStønadsperiode(stønadsperiode: Stønadsperiode): IkkeVurdert = this
+        override fun hentTidligesteDatoForAvslag(): LocalDate? = null
+        override fun erLik(other: Vilkår): Boolean {
+            return other is IkkeVurdert
+        }
+    }
+
+    data class Vurdert private constructor(
+        val vurderingsperioder: Nel<VurderingsperiodeFlyktning>,
+    ) : FlyktningVilkår() {
+
+        override val grunnlag: List<FlyktningGrunnlag> = vurderingsperioder.mapNotNull { it.grunnlag }
+        override fun lagTidslinje(periode: Periode): FlyktningVilkår {
+            return copy(
+                vurderingsperioder = Nel.fromListUnsafe(
+                    Tidslinje(
+                        periode = periode,
+                        objekter = vurderingsperioder,
+                    ).tidslinje,
+                ),
+            )
+        }
+
+        override val erInnvilget: Boolean = vurderingsperioder.all { it.resultat == Resultat.Innvilget }
+
+        override val erAvslag: Boolean = vurderingsperioder.any { it.resultat == Resultat.Avslag }
+
+        override val resultat: Resultat =
+            if (erInnvilget) Resultat.Innvilget else if (erAvslag) Resultat.Avslag else Resultat.Uavklart
+
+        override fun hentTidligesteDatoForAvslag(): LocalDate? {
+            return vurderingsperioder
+                .filter { it.resultat == Resultat.Avslag }
+                .map { it.periode.fraOgMed }
+                .minByOrNull { it }
+        }
+
+        override fun erLik(other: Vilkår): Boolean {
+            return other is Vurdert && vurderingsperioder.erLik(other.vurderingsperioder)
+        }
+
+        companion object {
+            fun tryCreate(
+                vurderingsperioder: Nel<VurderingsperiodeFlyktning>,
+            ): Either<UgyldigFlyktningVilkår, Vurdert> {
+                if (vurderingsperioder.overlappende()) {
+                    return UgyldigFlyktningVilkår.OverlappendeVurderingsperioder.left()
+                }
+                return Vurdert(vurderingsperioder).right()
+            }
+        }
+
+        sealed class UgyldigFlyktningVilkår {
+            object OverlappendeVurderingsperioder : UgyldigFlyktningVilkår()
+        }
+
+        override fun oppdaterStønadsperiode(stønadsperiode: Stønadsperiode): FlyktningVilkår {
+            check(vurderingsperioder.size == 1) { "Kan ikke oppdatere stønadsperiode for vilkår med med enn èn vurdering" }
```

Comment:
Burde denne heller ligge i tryCreate også gå via den?

Reply:
Da ville man vel i så fall måtte sperre for opprettelse av vilkår med flere vurderingsperioder? Det blir kanskje ikke helt som vi har tenkt oss (i hvert fall på sikt)? I denne sammenheng er vel denne sjekken mer en teknisk sperre "for ikke implementert enda" (hvis det i det hele tatt lar seg gjøre da)

Reply:
Prøvde svare via e-post; lurer på hvor det svaret havnet 😆  Men jeg skrev vel at vi kunne begrenset det til dette vilkåret. Men det er vel greit nok som det står. Dersom vi implementerer støtte for å dele opp dette i flere perioder i søknadsbehandlingen, og vi ikke tester å oppdatere stønadsperioden etter vi har valgt flere perioder vil vi få en liten overraskelse bare.

Reply:
Den sjekken ligger inne for alle vilkårene ser det ut som.  Godtar en liten overraskelse i framtiden.